go/vt/topo/etcd2topo/server.go

```diff
@@ -71,12 +82,42 @@ func (s *Server) Close() {
 	s.cli = nil
 }
 
-// NewServer returns a new etcdtopo.Server.
-func NewServer(serverAddr, root string) (*Server, error) {
-	cli, err := clientv3.New(clientv3.Config{
-		Endpoints:   strings.Split(serverAddr, ","),
+// TODO: Rename this to NewServer and change NewServer to a name that signifies it's global.
+func NewServerWithOpts(serverAddr, root string, certPath *string, keyPath *string, caPath *string) (*Server, error){
+	config := clientv3.Config{
+		Endpoints:  strings.Split(serverAddr, ","),
 		DialTimeout: 5 * time.Second,
-	})
+	}
+
+	// If TLS is enabled, attach TLS config info.
+	if *certPath != "" && *keyPath != "" {
+		// Verify we can load provided cert and key.
+		_, err := tls.LoadX509KeyPair(*certPath, *keyPath)
+		if err != nil {
+			log.Fatalf("Unable to load cert %v and key %v, err %v", *certPath, *keyPath, err)
+		}
+
+		// Verify we can open ca cert.
+		_, err = ioutil.ReadFile(*caPath)
+		if err != nil {
+			log.Fatalf("Unable to open ca cert %v, err %v", *caPath, err)
+		}
+
+		// Safe now to build up TLS info.
+		tlsInfo := transport.TLSInfo{
+			CertFile:   *certPath,
+			KeyFile:    *keyPath,
+			TrustedCAFile: *caPath,
+		}
+
+		tlsConfig, err := tlsInfo.ClientConfig()
+		if err != nil {
+			log.Fatalf("Unable to generate client config, err %v", err)
```

Comment:
Return an error instead of crashing.

Reply:
Updated https://github.com/vitessio/vitess/pull/5496/commits/245e974c99756dd2c00b17a874ba3cfc98b5a7f8.